src/components/progress/base.ts

```diff
@@ -0,0 +1,150 @@
+
+  @queryAssignedElements()
+  protected assignedElements!: Array<HTMLElement>;
+
+  @query('[part~="fill"]', true)
+  protected progressIndicator!: Element;
+
+  @state()
+  protected percentage = 0;
+
+  /** Maximum value of the control. */
+  @property({ type: Number })
+  public max = 100;
+
+  /** The value of the control. */
+  @property({ type: Number })
+  public value = 0;
+
+  /** The variant of the control. */
+  @property({ reflect: true })
+  public variant: 'primary' | 'info' | 'success' | 'warning' | 'danger' =
+    'primary';
+
+  /** Animation duration in milliseconds. */
+  @property({ type: Number, attribute: 'animation-duration' })
+  public animationDuration = 500;
+
+  /** The indeterminate state of the control. */
+  @property({ type: Boolean, reflect: false })
+  public indeterminate = false;
+
+  /** Shows/hides the label of the control. */
+  @property({ type: Boolean, attribute: 'hide-label', reflect: false })
+  public hideLabel = false;
+
+  /**
+   * Format string for the default label of the control.
+   * Placeholders:
+   *  {0} - current value of the control.
+   *  {1} - max value of the control.
+   */
+  @property({ attribute: 'label-format' })
+  public labelFormat!: string;
+
+  @watch('indeterminate', { waitUntilFirstUpdate: true })
+  protected indeterminateChange() {
+    if (this.indeterminate) {
+      this.cancelIndeterminate();
+    } else {
+      this.runAnimation(0, this.value);
+    }
+  }
+
+  @watch('max', { waitUntilFirstUpdate: true })
+  protected maxChange() {
+    this.max = Math.max(0, this.max);
+    if (this.value > this.max) {
+      this.value = this.max;
+    } else {
+      if (!this.indeterminate) {
+        this.runAnimation(0, this.value);
+      }
+    }
+  }
+
+  @watch('value', { waitUntilFirstUpdate: true })
+  protected valueChange(oldVal: number) {
+    this.value = clamp(this.value, 0, this.max);
+    if (!this.indeterminate) {
+      this.runAnimation(oldVal, this.value);
+    }
+  }
+
+  protected get animationOptions(): KeyframeAnimationOptions {
+    return {
+      easing: 'ease-out',
+      fill: 'forwards',
+      duration: this.animationDuration,
+    };
+  }
+
+  protected slotChanges() {
+    this.requestUpdate();
+  }
+
+  protected override firstUpdated() {
+    if (!this.indeterminate) {
+      this.runAnimation(0, this.value);
+    }
+  }
+
+  protected cancelIndeterminate() {
+    this.progressIndicator
+      ?.getAnimations()
+      .forEach((animation) => animation.cancel());
+  }
+
+  protected animateLabelTo(start: number, end: number) {
+    let t0: number;
+
+    const tick = (t1: number) => {
+      t0 = t0 ?? t1;
+
+      const progress = Math.min((t1 - t0) / (this.animationDuration || 1), 1);
+      this.percentage = Math.floor(
+        asPercent(progress * (end - start) + start, this.max)
+      );
+      progress < 1
+        ? (this.tick = requestAnimationFrame(tick))
+        : cancelAnimationFrame(requestAnimationFrame(tick));
```

Comment:
Because previously we didn't have a reference to the `tick` timestamp in the component itself.
Since we currently do, this can be changed to `cancelAnimationFrame(this.tick)`